pkg/chaincode/lifecycle.go

```diff
@@ -83,5 +88,12 @@ func (d *Definition) validate() error {
 	if d.Sequence <= 0 {
 		return fmt.Errorf("chaincode sequence must be greater than 0 for channel approve/commit")
 	}
+	if d.ApplicationPolicy != nil {
+		AppPolicydata, err := proto.Marshal(d.ApplicationPolicy)
+		if err != nil {
+			return err
+		}
+		d.validationParameter = AppPolicydata
+	}
```

Comment:
Having the `validationParameter` set as an (undocumented) side-effect of calling validate() seems potentially confusing. You could remove the `validationParameter` property and replace it with:

```go
func (d *Definition) validationParameter() ([]byte, error) {
	if d.ApplicationPolicy == nil {
		return nil, nil
	}

	appPolicyData, err := proto.Marshal(d.ApplicationPolicy)
	if err != nil {
		return nil, err
	}

	return appPolicyData, nil
}
```

What do you think?

Reply:
updated with a new implementation.